File: java/exp2/linearsearch.java
import java.util.*;
class linearsearch{
    public static void main(String args[]){
        Scanner s = new Scanner(System.in);
        int num,length;
        System.out.println("enter the length of array");
        length = s.nextInt();
        int[] arr = new int[length];
        System.out.println("enter the elements of the array");
        for (int i=0;i<length;i++){
            arr[i] =  s.nextInt();
        }
        System.out.println("enter the element to search");
        num = s.nextInt();
        int counter=0;
        for (int i=0;i<length;i++){
            if (arr[i]==num){
                System.out.println("the element is at index " + i);
                counter = 1;
            }
        }
        if (counter!=1){
            System.out.println("element not found in array");
        } 
    }
}

File: DSA/lab2/linearsearch.c
// ANSH RAJ
// 21BKT0116

#include <stdio.h>
int main(){
    int size;
    printf("enter the size :");
    int arr[size];
    scanf("%d",&size);
    printf("\nenter the elements :");
    for (int i=0;i<size;i++){
        scanf("%d",&arr[i]);
    }
    printf("enter the element to search");
    int key;
    scanf("%d",&key);
    for (int i=0;i<size;i++){
        if (arr[i]==key){
            printf("element %d found at index %d",key,i);
            break;
        }
        else{
            if (i==5){
                printf("search unsuccessful");
            }
        }
    }
return 0;}
